src/data/nat/prime.lean

```diff
@@ -976,3 +983,47 @@ namespace nat
 theorem prime_three : prime 3 := by norm_num
 
 end nat
+
+
+namespace nat
+
+/-- The only prime divisor of positive prime power `p^k` is `p` itself -/
+lemma prime_pow_prime_divisor {p k : ℕ} (hk : 0 < k) (hp: prime p) :
+  (p^k).factors.to_finset = {p} :=
+begin
+  rw hp.factors_pow, exact list.to_finset_repeat_of_pos hk,
+end
+
+lemma factors_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) (p : ℕ) :
+  p ∈ (a * b).factors ↔ p ∈ a.factors ∨ p ∈ b.factors :=
+begin
+  rw [mem_factors (mul_pos ha hb), mem_factors ha, mem_factors hb, ←and_or_distrib_left],
+  simp only [and.congr_right_iff], exact prime.dvd_mul,
+end
+
+/-- If `a`,`b` are positive the prime divisors of `(a * b)` are the union of those of `a` and `b` -/
+lemma prime_divisors_mul_of_pos {a b : ℕ} (ha : 0 < a) (hb : 0 < b) :
+  (a * b).factors.to_finset = a.factors.to_finset ∪ b.factors.to_finset :=
+begin
+  ext p, rw finset.mem_union, repeat {rw list.mem_to_finset}, exact factors_mul_of_pos ha hb p,
+end
```

Comment:
```suggestion
by { ext p, simp only [finset.mem_union, list.mem_to_finset, factors_mul_of_pos ha hb p] }
```

for example, this is a `by` block so it's more ok. also, `repeat rw` is pretty much always a worse version of `simp only`